### PR TITLE
Dssat

### PR DESCRIPTION
**Pull request by cyrillemidingoyi**
Hello
Please, review the implementation of dssat example in java, cython and fortran that I have proposed
Cyrille

### CODE REVIEW

File: example/DSSAT/CropML_PETET_sample.xml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+<Input name="TMAX" description="Maximum daily temperature (�C)" datatype="DOUBLE" min="-50" max="60" default="20" unit="http://www.wurvoc.org/vocabularies/om-1.8/degree_Celsius" inputtype="variable" variablecategory="auxiliary" />
+<Input name="TMIN" description="Minimum daily temperature (�C)" datatype="DOUBLE" min="-50" max="60" default="10" unit="http://www.wurvoc.org/vocabularies/om-1.8/degree_Celsius" inputtype="variable" variablecategory="auxiliary" />
+<Input name="XHLAI" description="Leaf area index (m2[leaf] / m2[ground])" datatype="DOUBLE" min="0" max="10" default="3" unit="http://www.wurvoc.org/vocabularies/om-1.8/one" inputtype="variable" variablecategory="auxiliary" />
+</Inputs>
+<Outputs>
+<Output name="EO" description="Potential evapotranspiration rate (mm/d)" datatype="DOUBLE" min="0.0" max="1.0" default="0.0" unit="http://www.wurvoc.org/vocabularies/om-1.8/one" variablecategory="auxiliary" />
+</Outputs>
+<Algorithm language="Java" platform="DSSAT" filename="PETPT.java" function="PETPT">
```
Comment (pradal) [inline example/DSSAT/CropML_PETET_sample.xml]: if the language is Java, the platform is not DSSAT

File: example/DSSAT/CropML_PETET_sample.xml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ELSE IF (TMAX .LT. 5.0) THEN
+	EO = EEQ*0.01*EXP(0.18*(TMAX+20.0))
+ENDIF
+!###  EO = MAX(EO,0.0)   !gives error in DECRAT_C
+EO = MAX(EO,0.0001)
+]]>
+</Algorithm>
+<Algorithm language=Cython" platform="DSSAT" filename="PETPT.pyx" function="PETPT">
+<![CDATA[
+cdef double ALBEDO, EEQ, SLANG, TD
```
Comment (pradal) [inline example/DSSAT/CropML_PETET_sample.xml]: from math import exp

File: example/DSSAT/CropML_PETET_sample.xml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+TD = 0.60*TMAX + 0.40*TMIN
+if (XHLAI <= 0.0):
+    ALBEDO = MSALB
+else:
+    ALBEDO = 0.23 - (0.23-MSALB)*math.exp(-0.75*XHLAI)
```
Comment (pradal) [inline example/DSSAT/CropML_PETET_sample.xml]: replace math.exp by exp

File: example/DSSAT/CropML_PETET_sample.xml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+SLANG = SRAD * 23.923
+EEQ = SLANG * (2.04E-4 - 1.83E-4 * ALBEDO)*(TD+29.0);
+eo = EEQ*1.1
+if (TMAX < 35.0):
+    eo = EEQ*((TMAX-35.0)*0.05+1.1
+else:
+    eo =  EEQ*0.01*math.exp(0.18*(TMAX+20.0);
```
Comment (pradal) [inline example/DSSAT/CropML_PETET_sample.xml]: replace math.exp by exp

File: example/DSSAT/CropML_PETET_sample.xml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+#EO = MAX(EO,0.0)   !gives error in DECRAT_C
```
Comment (pradal) [inline example/DSSAT/CropML_PETET_sample.xml]: Remove the line #E0
Use max instead of math.max

File: example/DSSAT/CropML_PETET_sample_link.xml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+<Input name="XHLAI" description="Leaf area index (m2[leaf] / m2[ground])" datatype="DOUBLE" min="0" max="10" default="3" unit="http://www.wurvoc.org/vocabularies/om-1.8/one" inputtype="variable" variablecategory="auxiliary" />
+</Inputs>
+<Outputs>
+<Output name="EO" description="Potential evapotranspiration rate (mm/d)" datatype="DOUBLE" min="0.0" max="1.0" default="0.0" unit="http://www.wurvoc.org/vocabularies/om-1.8/one" variablecategory="auxiliary" />
+</Outputs>
+<Algorithm language="Java" platform="DSSAT" filename="PETPT.java" function="PETPT">
+</Algorithm>
+<Algorithm language="Fortran" platform="DSSAT" filename="PETPT.f" function="PETPT">
+</Algorithm>
+<Algorithm language="Cython" platform="DSSAT" filename="PETPT.pyx" function="PETPT">
```
Comment (pradal) [inline example/DSSAT/CropML_PETET_sample_link.xml]: Use platform appropriately.
DSSAT is only for the Fortran code. The rest is either no platform, or SimPlace platform for Java.
You can also add the C# version with platform BioMa.

File: example/DSSAT/PETPT.f [hunk 7]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (pradal) [inline example/DSSAT/PETPT.f]: Here, you need to add the header (subroutine PEPT)

File: example/DSSAT/PETPT.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
+double ALBEDO;
```
Comment (pradal) [inline example/DSSAT/PETPT.java]: This is not Java. You need to create a function. Something like

public static double methodName(...) {
   // body
}

File: example/DSSAT/PETPT.pyx [hunk 9]
```diff
@@ -0,0 +1,16 @@
+cdef double ALBEDO, EEQ, SLANG, TD
```
Comment (pradal) [inline example/DSSAT/PETPT.pyx]: Same here:

def PETP(....):

File: example/DSSAT/original/PETPT_DSSAT.f [hunk 10]
```diff
@@ -0,0 +1,315 @@
+C=======================================================================
```
Comment (pradal) [inline example/DSSAT/original/PETPT_DSSAT.f]: This is exactly the code you have to give in Fortran.

Comment (cyrillemidingoyi) [reply example/DSSAT/original/PETPT_DSSAT.f]: I thought that it just the algorithm we will put in the file. Ok I will do it

File: example/DSSAT/CropML_PETET_sample.xml [hunk 11]
```diff
@@ -16,29 +16,28 @@
...
-}
+		double ALBEDO;
+		double SLANG;
+		double EEQ;
+		double eo;
+		double TD = 0.60*TMAX + 0.40*TMIN;
+		if (XHLAI <= 0.0) {
+                    ALBEDO = MSALB;
+		} else {
+			ALBEDO = 0.23 - (0.23-MSALB)*Math.exp(-0.75*XHLAI);
```
Comment (pradal) [inline example/DSSAT/CropML_PETET_sample.xml]: Use the same ident than the previous statement

File: example/DSSAT/CropML_PETET_sample.xml [hunk 12]
```diff
@@ -16,29 +16,28 @@
...
+		}
+		SLANG = SRAD * 23.923;
+		EEQ = SLANG * (2.04E-4 - 1.83E-4 * ALBEDO)*(TD+29.0);
+		eo = EEQ*1.1;
+		if (TMAX < 35.0) {
+            eo = EEQ*((TMAX-35.0)*0.05+1.1);
```
Comment (pradal) [inline example/DSSAT/CropML_PETET_sample.xml]: Do not mix tab and spaces

File: example/DSSAT/CropML_PETET_sample_link.xml [hunk 13]
```diff
@@ -16,11 +16,13 @@
 <Outputs>
 <Output name="EO" description="Potential evapotranspiration rate (mm/d)" datatype="DOUBLE" min="0.0" max="1.0" default="0.0" unit="http://www.wurvoc.org/vocabularies/om-1.8/one" variablecategory="auxiliary" />
 </Outputs>
-<Algorithm language="Java" platform="DSSAT" filename="PETPT.java" function="PETPT">
+<Algorithm language="Java" platform="" filename="PETPT.java" function="PETPT">
 </Algorithm>
-<Algorithm language="Fortran" platform="DSSAT" filename="PETPT.f" function="PETPT">
```
Comment (pradal) [inline example/DSSAT/CropML_PETET_sample_link.xml]: Here, you can st platform="DSSAT"

File: example/DSSAT/PETPT.f [hunk 14]
```diff
@@ -16,3 +19,7 @@
 ENDIF
 !###  EO = MAX(EO,0.0)   !gives error in DECRAT_C
 EO = MAX(EO,0.0001)
+
+RETURN
```
Comment (pradal) [inline example/DSSAT/PETPT.f]: I do not remember if you have to RETURN something in Fortran.

Comment (cyrillemidingoyi) [reply example/DSSAT/PETPT.f]: yes, it is also in the original file.

File: example/DSSAT/original/PETPT_DSSAT.f [hunk 15]
```diff
@@ -0,0 +1,315 @@
...
+
+C  PETPT, Subroutine, J.T. Ritchie
```
Comment (pradal) [inline example/DSSAT/original/PETPT_DSSAT.f]: It seems to me that the code is duplicated. Isn't it?

File: example/DSSAT/PETPT.f (hunk 14 above)
Comment (pradal) [reply example/DSSAT/PETPT.f]: OK. Nice.

File: example/DSSAT/CropML_PETET_sample_link.xml (hunk 13 above)
Comment (cyrillemidingoyi) [reply example/DSSAT/CropML_PETET_sample_link.xml]: Please, i don't understand "st".

Comment (pradal) [reply example/DSSAT/CropML_PETET_sample_link.xml]: st -> set
Sorry

File: example/DSSAT/original/PETPT_DSSAT.f (hunk 15 above)
Comment (cyrillemidingoyi) [reply example/DSSAT/original/PETPT_DSSAT.f]: Yes, it is. I correct it.

**Comment by pradal** [comment]
Close #11

**Pull request merged**